### ISSUE TITLE
False positive error when using  `proc_macro` crate

### ISSUE THREAD

**Issue by la10736** [intellij-rust/intellij-rust#2364]
```rust
#![feature(proc_macro)]

extern crate proc_macro;
```
produce a false _Can't find crate for `proc_macro` [E0463]_ error highlight.
![schermata da 2018-03-11 11-49-51](https://user-images.githubusercontent.com/3902961/37252554-e3b92e30-2522-11e8-8858-5e10524be89e.png)

**Comment by ELD** [comment]
Yep, I get this too. Additionally, it doesn't supply any completions for types coming out of the `proc_macro` crate. Supporting this would be quite nice to have.

**Comment by vlad9486** [comment]
The problem is [here](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/cargo/project/workspace/StandardLibrary.kt#L43)
The `findFileByRelativePath` method returns `null` because the library `proc_macro` is not part of `rust-src` distribution as pointed [here](https://github.com/japaric/xargo/issues/125#issuecomment-325204329), so my solution is clone rust sources and create soft link. The command is 
`$ln -s ~/projects/rust/src/libproc_macro ~/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/src/`
Your paths might be different.

**Comment by ELD** [comment]
@vlad20012 Oh, I didn't realize that the `proc_macro` source didn't come with the `rust-src` distribution. Good to know. Thanks for the workaround! 👍

**Comment by Undin** [comment]
Now `rust-src` component for the latest nightly rust contains `proc-macro` sources so I think it can be closed
